storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArrayScoreCalculator.java

```diff
@@ -97,14 +101,19 @@ private static void computeDelta(
     // justified state to a new state with a higher epoch that is on a different fork
     // because that may have on-boarded less validators than the prior fork.
     UInt64 newBalance =
-        newBalances.size() > validatorIndexInt ? newBalances.get(validatorIndexInt) : UInt64.ZERO;
+        newBalances.size() > validatorIndexInt && !vote.isNextEquivocating()
+            ? newBalances.get(validatorIndexInt)
+            : UInt64.ZERO;
 
     if (!vote.getCurrentRoot().equals(vote.getNextRoot()) || !oldBalance.equals(newBalance)) {
       subtractBalance(getIndexByRoot, deltas, vote.getCurrentRoot(), oldBalance);
       addBalance(getIndexByRoot, deltas, vote.getNextRoot(), newBalance);
-
-      VoteTracker newVote =
-          new VoteTracker(vote.getNextRoot(), vote.getNextRoot(), vote.getNextEpoch());
+      final VoteTracker newVote;
+      if (vote.isNextEquivocating()) {
+        newVote = VoteTracker.markCurrentEquivocating(vote);
+      } else {
+        newVote = VoteTracker.create(vote.getNextRoot(), vote.getNextRoot(), vote.getNextEpoch());
+      }
```

Comment:
I'd just simplify this to:
```suggestion
      newVote = new VoteTracker(vote.getNextRoot(), vote.getNextRoot(), vote.getNextEpoch(), vote.isNextEquivocating(), vote.isNextEquivocating());
```

it probably doesn't matter if we don't actually update the current roots once the validator is equivocating but since it's simple, it makes sense to get our records straight and show that we have considered that next vote now.

Reply:
Returning to this one. I've applied your suggestion but according to the spec, we shouldn't update votes for equivocated validators. This shouldn't hurt anything from the first look, but it could be our bug in the future

Reply:
I've reverted this one too, but if you think it's safe to update it, I will put it back